src/chatgpt.rs

```diff
@@ -98,8 +124,92 @@ impl fmt::Display for Role {
     }
 }
 
+fn ensure_config_file(dir: &PathBuf) -> std::io::Result<PathBuf> {
+    fs::create_dir_all(dir)?;
+    let config_path = dir.join("config.toml");
+    if !config_path.exists() {
+        let config = AppConfig::default();
+        let contents = toml::to_string(&config).expect("Failed to serialize config");
+        let mut file = File::create(&config_path)?;
+        file.write_all(contents.as_bytes())?;
+    }
+
+    Ok(config_path)
+}
+
 impl GptClient {
+    pub fn setup_config(dir: &PathBuf) {
+        if let Err(e) = ensure_config_file(dir) {
+            panic!("Failed to ensure config file exists: {}", e);
+        }
+    }
+
+    pub fn load_config(dir: &PathBuf) -> AppConfig {
```

Comment:
🤔 perhaps config related functions deserve their own file ? I' also think that if you aren't referencing self in the functions they don't need to be part of the implementation of a struct either, they can  just be loose functions.

Reply:
I'm referencing self in the get and set (just not the setup and load) so that I don't have to redo defining the config file location whenever I want to get the values ... seeing how I can rework

Reply:
Was able to restructure this in a way that make sense

Reply:
Awesome, great work!